### PR TITLE
Domains: Remove unused `Edit` component and related subcomponents

### PR DESCRIPTION
**Pull request by leonardost**
#### Proposed Changes

This PR removes the unused `Edit` component and its subcomponents. They were deprecated when we redesigned the domain management page (see pcYYhz-m2-p2) at the end of last year/start of this one but remained in the code until now. #60096 was the PR that switched the domain management page to the new `Settings` component.

For reference, this was what the `Edit` component looked like:

Screen Shot 2022-09-13 at 12 00 37

There were some subcomponents whose functionality might not have been migrated to the new `Setting` component and which might still be useful (`MaintenanceCard`, `ExpiringCreditCard`, `ExpiringSoon`, `DomainExpiryOrRenewal`, and others) so they'll be removed in a later PR (also to keep this PR's size under control).

#### Testing Instructions

- Code inspection should be enough as these were all unused components, but please take a look at the domain settings pages to ensure everything looks alright

#### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

### CODE REVIEW

**Comment by gius80** [comment]
Thanks for cleaning up! Just a minor comment: we need to import `styles.scss` somewhere because it was imported in one of the deleted files and we lost a couple of css rules (such as the domain settings background that now is gray).

Markup on 2022-09-19 at 11:23:09

**Comment by leonardost** [comment]
Great catch @gius80, thank you for noticing that! I'll import that file in one of the used components

File: client/my-sites/domains/domain-management/settings/style.scss [hunk 1]
```diff
@@ -2,6 +2,10 @@
 @import "@wordpress/base-styles/breakpoints";
 @import "@wordpress/base-styles/mixins";
 
+body.edit__body-white.theme-default.color-scheme {
+	--color-surface-backdrop: var(--studio-white);
+}
```
Comment (leonardost) [inline client/my-sites/domains/domain-management/settings/style.scss:7]: This is the only bit of `/edit/style.scss` that was being used

Comment (gius80) [reply client/my-sites/domains/domain-management/settings/style.scss:7]: Super! Now it's ready to go 🚀🚀🚀

**Review by gius80 — APPROVED** [review]

**Pull request merged**